controller.py

```diff
@@ -378,17 +378,17 @@ def index(page=1):
     # do a search if you have a search term
     # (make this more general for an all fields search)
     if s:
-        books = Book.query.order_by(Book.title.asc()).filter(or_(Book.title.contains(s), Book.authors.contains(s), Book.subjects.contains(s))).paginate(page,PAGINATE_BY_HOWMANY,False)
+        books = Book.query.order_by(Book.title.asc()).filter(or_(Book.title.contains(s), Book.authors.contains(s), Book.subjects.contains(s))).paginate(page=page, per_page=PAGINATE_BY_HOWMANY, error_out=False)
 
     # return all books, currently sort by title ascending.
     else:
-        books = Book.query.order_by(Book.title.asc()).paginate(page,PAGINATE_BY_HOWMANY,False)
+        books = Book.query.order_by(Book.title.asc()).paginate(page=page, per_page=PAGINATE_BY_HOWMANY, error_out=False)
 
     return render_template('index.html', books=books, s=s)
 
 if __name__ == "__main__":
     # flask can execute arbitrary python if you do this.
     # app.run(host='0.0.0.0') # listens on all public IPs. 
 
-    app.run()
+    app.run(host='0.0.0.0')
```

Comment:
Caddy fronts this with a reverse proxy

```suggestion
    app.run(host='127.0.0.1')
```

Reply:
It needs to be `0.0.0.0` otherwise you can't port forward into docker.

Reply:
Added a section to the config file.